diff --git a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/Screen.kt b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/Screen.kt
index 2b7dcff..e263012 100644
--- a/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/Screen.kt
+++ b/node_modules/react-native-screens/android/src/main/java/com/swmansion/rnscreens/Screen.kt
@@ -16,6 +16,7 @@ import androidx.core.view.children
 import androidx.fragment.app.Fragment
 import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
 import com.facebook.react.bridge.GuardedRunnable
+import com.facebook.react.bridge.UiThreadUtil
 import com.facebook.react.bridge.ReactContext
 import com.facebook.react.uimanager.PixelUtil
 import com.facebook.react.uimanager.ThemedReactContext
@@ -488,6 +489,10 @@ class Screen(
 
     private fun startTransitionRecursive(parent: ViewGroup?) {
         parent?.let {
+            if (!UiThreadUtil.isOnUiThread()) {
+                UiThreadUtil.runOnUiThread { startTransitionRecursive(parent) }
+                return
+            }
             for (i in 0 until it.childCount) {
                 val child = it.getChildAt(i)
 
